server/lib/pbench/__init__.py

```diff
@@ -153,16 +154,25 @@ def get_pbench_logger(caller, config):
         except FileExistsError:
             # directory already exists, ignore
             pass
-        fh = logging.FileHandler(os.path.join(logdir, "{}.log".format(caller)))
-        fh.setLevel(logging.DEBUG)
+
+        if config.logger_type == "file":
+            handler = logging.FileHandler(os.path.join(logdir, "{}.log".format(caller)))
+        elif config.logger_type == "devlog":
+            handler = logging.SyslogHandler(address='/dev/log', facility=logging.handlers.SysLogHandler.LOG_DAEMON)
+        elif config.logger_type == "hostport":
+            handler = logging.SysLogHandler(address=(config.logger_host, int(config.logger_port)))
```

Comment:
I think we want to use TCP, so you will need to add a `socktype=socket.SOCK_STREAM ` here,  since the default is UDP (see https://docs.python.org/3/library/logging.handlers.html#sysloghandler).

Reply:
ok sure

Reply:
Well, famous last words - running some experiments with SOCK_STREAM has uncovered problems: apparently the Python logging module's implementation of the SOCK_STREAM interface is buggy.

So please revert this change: delete the `socktype` option and just let it default to UDP.

Reply:
@riya-17, let's add a comment, here, and in `pbench-server-default.cfg`, that the `hostport` logger type is UDP based logging.